test/unit/core/bit_manipulation_test.cpp

```diff
@@ -108,6 +108,103 @@ TYPED_TEST(unsigned_operations, bit_scan_reverse)
     }
 }
 
+TYPED_TEST(unsigned_operations, count_leading_zeros)
+{
+    using unsigned_t = TypeParam;
+    constexpr size_t t1 = count_leading_zeros<unsigned_t>(0b0001);
+    constexpr size_t t2 = count_leading_zeros<unsigned_t>(0b0101);
+    constexpr size_t t3 = count_leading_zeros<unsigned_t>(0b0010);
+    constexpr size_t t4 = count_leading_zeros<unsigned_t>(0b0110);
+    constexpr size_t t5 = count_leading_zeros<unsigned_t>(0b0100);
+    constexpr size_t t6 = count_leading_zeros<unsigned_t>(0b10100000);
+    EXPECT_EQ(t1, sizeof_bits<unsigned_t> - 1u);
+    EXPECT_EQ(t2, sizeof_bits<unsigned_t> - 3u);
+    EXPECT_EQ(t3, sizeof_bits<unsigned_t> - 2u);
+    EXPECT_EQ(t4, sizeof_bits<unsigned_t> - 3u);
+    EXPECT_EQ(t5, sizeof_bits<unsigned_t> - 3u);
+    EXPECT_EQ(t6, sizeof_bits<unsigned_t> - 8u);
+
+    for (uint8_t cnt = 0; cnt < sizeof_bits<unsigned_t>; ++cnt)
+    {
+        unsigned_t start = std::numeric_limits<unsigned_t>::max() >> cnt;
+        unsigned_t end = start >> 1u;
+        for (unsigned_t n = start, k = 0u; n < end && k < max_iterations; ++n, ++k)
+        {
+            EXPECT_EQ(count_leading_zeros(n), cnt) << "n " << n << " should have " << cnt << " leading zeros.";
```

Comment:
can you add the sdsl versions. They might behave differently and we will never know, because we can't test them.